extension/src/popup/components/WarningMessages/index.tsx

```diff
@@ -307,28 +315,34 @@ export const ScamAssetWarning = ({
       .build()
       .toXDR();
 
-    const res = await dispatch(
-      signFreighterTransaction({
-        transactionXDR,
-        network: networkDetails.networkPassphrase,
-      }),
-    );
-
-    if (signFreighterTransaction.fulfilled.match(res)) {
-      const submitResp = await dispatch(
-        submitFreighterTransaction({
-          signedXDR: res.payload.signedTransaction,
-          networkDetails,
+    if (isHardwareWallet) {
```

Comment:
very similar to the code in [ManageAssetRows.index](https://github.com/stellar/freighter/blob/9174d0a088312d53e8b11a84e739e2a9c34335c3/extension/src/popup/components/manageAssets/ManageAssetRows/index.tsx#L140-L145)